### PR TITLE
Dev/twegner/issue 9

### PR DESCRIPTION
**Pull request by trwegner**
Change Annotation to Event for #9.  Added XML comment for publics in touched files to address [CS1591](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/cs1591) warnings.  Fixed a few typos too.

### CODE REVIEW

**Comment by SergeyKanzhelev** [comment]
@trwegner great! It's OK to eliminate IMessageEvent as a separate PR. Just create an issue and I will merge this one. It looks good

**Comment by trwegner** [comment]
I've updated #16 to include  open-telemetry/opentelemetry-java#13 and open-telemetry/opentelemetry-java#141.

**Comment by lmolkova** [comment]
LGTM

File: src/OpenTelemetry.Exporter.Zipkin/Implementation/ZipkinSpan.cs [hunk 1]
```diff
@@ -51,8 +51,8 @@ internal class ZipkinSpan
         [JsonProperty("remoteEndpoint")]
         public ZipkinEndpoint RemoteEndpoint { get; set; }
 
-        [JsonProperty("annotations")]
-        public List<ZipkinAnnotation> Annotations { get; set; }
```
Comment (SergeyKanzhelev) [inline src/OpenTelemetry.Exporter.Zipkin/Implementation/ZipkinSpan.cs]: zipkin is still using name "annotation". This file represents Zipkin's schema, not OpenTelemetry one. Please revert the change in this file.

We really need tests for Zipkin to catch it in unit tests =)

File: src/OpenTelemetry.Abstractions/Trace/Config/TraceParams.cs [hunk 2]
```diff
@@ -26,24 +26,24 @@ public sealed class TraceParams : ITraceParams
...
         /// </summary>
         public static readonly ITraceParams Default =
-            new TraceParams(Samplers.GetProbabilitySampler(DefaultProbability), DefaultSpanMaxNumAttributes, DefaultSpanMaxNumAnnotations, DefaultSpanMaxNumMessageEvents, DefaultSpanMaxNumLinks);
+            new TraceParams(Samplers.GetProbabilitySampler(DefaultProbability), DefaultSpanMaxNumAttributes, DefaultSpanMaxNumEvents, DefaultSpanMaxNumMessageEvents, DefaultSpanMaxNumLinks);
 
         private const double DefaultProbability = 1e-4;
         private const int DefaultSpanMaxNumAttributes = 32;
-        private const int DefaultSpanMaxNumAnnotations = 32;
+        private const int DefaultSpanMaxNumEvents = 32;
         private const int DefaultSpanMaxNumMessageEvents = 128;
```
Comment (SergeyKanzhelev) [inline src/OpenTelemetry.Abstractions/Trace/Config/TraceParams.cs]: if you are removing message events - this parameter (`DefaultSpanMaxNumMessageEvents `) doesn't make sense any longer

File: src/OpenTelemetry.Abstractions/Trace/Export/ISpanData.cs [hunk 3]
```diff
@@ -44,47 +44,47 @@ public interface ISpanData
...
+        /// Gets the collection of <see cref="ITimedEvents{IEvent}"/> objects.
         /// </summary>
-        ITimedEvents<IAnnotation> Annotations { get; }
+        ITimedEvents<IEvent> Events { get; }
 
         /// <summary>
-        /// Gets the collection of message events.
+        /// Gets the collection of <see cref="ITimedEvents{IMessageEvent}"/> objects.
         /// </summary>
         ITimedEvents<IMessageEvent> MessageEvents { get; }
```
Comment (SergeyKanzhelev) [inline src/OpenTelemetry.Abstractions/Trace/Export/ISpanData.cs]: I thought you are removing it. I'm confused =)

Comment (trwegner) [reply src/OpenTelemetry.Abstractions/Trace/Export/ISpanData.cs]: From the suggestion last night I was planning to create a follow up PR for removing MessageEvents to address #16 and keep this PR fucused on #9 . I'll push a change to revert the ZipKin changes soon.

Comment (trwegner) [reply src/OpenTelemetry.Abstractions/Trace/Export/ISpanData.cs]: Okay, reverted Zipkin changes and will plan to work on #16 after this PR is complete.

File: src/OpenTelemetry.Abstractions/Trace/Config/TraceParams.cs (hunk 2 above)
Comment (trwegner) [reply src/OpenTelemetry.Abstractions/Trace/Config/TraceParams.cs]: Plan to address this in a separate PR for #16

**Comment by SergeyKanzhelev** [comment]
build fails with:

```
Implementation\TraceExporterHandler.cs(100,49): error CS1061: 'ISpanData' does not contain a definition for 'Annotations' and no accessible extension method 'Annotations' accepting a first argument of type 'ISpanData' could be found (are you missing a using directive or an assembly reference?) [D:\a\1\s\src\OpenTelemetry.Exporter.Zipkin\OpenTelemetry.Exporter.Zipkin.csproj]
Implementation\TraceExporterHandler.cs(100,49): error CS1061: 'ISpanData' does not contain a definition for 'Annotations' and no accessible extension method 'Annotations' accepting a first argument of type 'ISpanData' could be found (are you missing a using directive or an assembly reference?) [D:\a\1\s\src\OpenTelemetry.Exporter.Zipkin\OpenTelemetry.Exporter.Zipkin.csproj]
```

**Comment by trwegner** [comment]
Pushed update for src\OpenTelemetry.Exporter.Zipkin\Implementation\TraceExporterHandler.cs to address the break.

**Comment by SergeyKanzhelev** [comment]
```
TestZipkin.cs(80,22): error CS1061: 'ISpan' does not contain a definition for 'AddAnnotation' and no accessible extension method 'AddAnnotation' accepting a first argument of type 'ISpan' could be found (are you missing a using directive or an assembly reference?) [D:\a\1\s\src\Samples\Samples.csproj]
```

**Comment by trwegner** [comment]
With last push all checks are passing now.
